src/util/snippets.py

```diff
@@ -243,16 +249,26 @@ def _dumps(self, node):
 
 
 class Convertor:
-    def __init__(self, converters, callback):
+    def __init__(self, converters, callback, dbname, update_query):
         self.converters = converters
         self.callback = callback
+        self.dbname = dbname
+        self.update_query = update_query
+
+    def __call__(self, query):
+        with db_connect(self.dbname).cursor() as cr:
+            cr.execute(query)
+            for self.row in cr.fetchall():
+                self._convert_row()
+                if "id" in self.row:
+                    cr.execute(self.update_query, self.row)
```

Comment:
I'd say the idea is to pass the row and return the changes, if any.
```suggestion
            for changes in filter(None, map(self._convert_row, cr.fetchall())):
                cr.execute(self.update_query, changes)
```

Reply:
> I'd say the idea is to pass the row and return the changes, if any.

That's what I meant by "procedural". Passing data back and forth through arguments and return values between two methods of a class feels just icky.

Reply:
> That's what I meant by "procedural". Passing data back and forth through arguments and return values between two methods of a class feels just icky.

You can declare the method `@staticmethod` or `@classmethod` if that feels better. The possibility to override the processing is an advantage of keeping it a method. The latter is not necessary right now. To me the best value here is to clearly separate what is actually processing _one_ row, vs the underlying machinery to fetch/send data from/to the DB. The simpler diff is just a nice thing.